src/main/scala/lc/background/taskThread.scala

```diff
@@ -22,13 +22,21 @@ class BackgroundTask(throttle: Int, timeLimit: Int) {
         if (imageNum.next())
           throttleIn = (throttleIn - imageNum.getInt("total"))
         while (0 < throttleIn) {
-          Captcha.generateChallenge(Parameters("medium", "image/png", "text", Option(Size(0, 0))))
+          Captcha.generateChallenge(getRandomParam())
           throttleIn -= 1
         }
       } catch { case exception: Exception => println(exception) }
     }
   }
 
+  private def getRandomParam(): Parameters = {
+    val level = Config.allowedLevels.toList(Config.getNextRandomInt(Config.allowedLevels.size))
+    val media = Config.allowedMedia.toList(Config.getNextRandomInt(Config.allowedMedia.size))
+    val inputType = Config.allowedInputType.toList(Config.getNextRandomInt(Config.allowedInputType.size))
+
+    Parameters(level, media, inputType, Some(Size(0,0)))
```

Comment:
Won't this create invalid combinations? If two different captcha providers support `(easy, png)` and `(medium, gif)` then this might create an invalid combination `(easy, gif)`. What happens in that case?

Reply:
Good catch. I'll pick a captcha provider by random instead of just params. We can then choose allowed params by the captcha provider.

Reply:
yup, that's perfect